crates/circuit/src/dag_circuit.rs

```diff
@@ -3335,11 +3336,27 @@ def _format(operand):
     ///     Ipython.display.Image: if in Jupyter notebook and not saving to file,
     ///     otherwise None.
     #[pyo3(signature=(scale=0.7, filename=None, style="color"))]
-    fn draw(&self, scale: f64, filename: Option<&str>, style: &str) -> PyResult<Py<PyAny>> {
-        // from qiskit.visualization.dag_visualization import dag_drawer
-        //
-        // return dag_drawer(dag=self, scale=scale, filename=filename, style=style)
-        todo!()
+    fn draw<'py>(
+        &self,
+        py: Python<'py>,
+        scale: f64,
+        filename: Option<&str>,
+        style: &str,
+    ) -> PyResult<Bound<'py, PyAny>> {
+        let module = PyModule::import_bound(py, "qiskit.visualization.dag_visualization")?;
+        module.call_method1("dag_drawer", (scale, filename, style))
```

Comment:
Don't you need to pass the `dag` to this method call somehow?

Reply:
Oh yeah self should be the first argument here. I'm not sure how this worked. I definitely generated an image using `dag.draw()` in my local testing. I'll fix this and verify my test is actually testing what I thought it was.

Reply:
Ok this should be fixed now with https://github.com/kevinhartman/qiskit/pull/22/commits/b42d0ac7275ffc007940fcac9d3ffa1ff2e2f51c 

I also ran the test script: 

```python
from qiskit import QuantumCircuit
from qiskit.converters import circuit_to_dag
from qiskit.visualization import dag_drawer
qc = QuantumCircuit(2)
qc.h(0)
qc.cx(0, 1)
qc.cz(1, 0)
qc.measure_all()
dag = circuit_to_dag(qc)
print(dag)
print(type(dag))
#dag_drawer(dag, filename='/tmp/test.png')

dag.draw(filename="/tmp/test_new.png")
```

which output:
```
<qiskit._accelerate.circuit.DAGCircuit object at 0x5cace27c5cd0>
<class 'qiskit._accelerate.circuit.DAGCircuit'>
```
and generated the following image 

![test_new](https://github.com/kevinhartman/qiskit/assets/2447371/e489040e-83b7-411c-afe7-689b5fe57230)

Also, locally I added a `println!` to the draw method rust code to assert it was actually running the method which was also displayed. So I am fairly confident now that it's working correctly.